core/trino-main/src/test/java/io/trino/sql/planner/assertions/WindowFunctionProvider.java

```diff
@@ -27,26 +34,42 @@ final class WindowFunctionProvider
     private final String name;
     private final WindowNode.Frame frame;
     private final List<PlanTestSymbol> args;
+    private final List<PlanMatchPattern.Ordering> orderBy;
 
-    public WindowFunctionProvider(String name, WindowNode.Frame frame, List<PlanTestSymbol> args)
+    public WindowFunctionProvider(String name, WindowNode.Frame frame, List<PlanTestSymbol> args, List<PlanMatchPattern.Ordering> orderBy)
     {
         this.name = requireNonNull(name, "name is null");
         this.frame = requireNonNull(frame, "frame is null");
         this.args = requireNonNull(args, "args is null");
+        this.orderBy = ImmutableList.copyOf(orderBy);
     }
 
     @Override
     public String toString()
     {
-        return "%s(%s) %s".formatted(
+        return "%s(%s%s) %s".formatted(
                 name,
                 Joiner.on(", ").join(args),
+                orderBy.isEmpty() ? "" : " ORDER BY " + Joiner.on(", ").join(orderBy),
                 frame);
     }
 
     @Override
     public WindowFunction getExpectedValue(SymbolAliases aliases)
     {
-        return new WindowFunction(name, frame, toSymbolReferences(args, aliases));
+        Optional<OrderingScheme> orderByClause = Optional.empty();
+        if (!orderBy.isEmpty()) {
+            ImmutableList.Builder<Symbol> fields = ImmutableList.builder();
+            ImmutableMap.Builder<Symbol, SortOrder> orders = ImmutableMap.builder();
+
+            for (PlanMatchPattern.Ordering ordering : this.orderBy) {
+                Symbol symbol = new Symbol(UNKNOWN, aliases.get(ordering.getField()).name());
```

Comment:
UNKNOWN is incorrect. It should use the type of the field.

Reply:
I copied from `AggregationFunctionProvider`; annotate says it was yours :P.
Fixed both

Reply:
Indeed it was. It was during the migration to add types to IR expressions :)